sktime/forecasting/base/convertIO/_convertIO.py

```diff
@@ -0,0 +1,190 @@
+# -*- coding: utf-8 -*-
+"""
+Machine type converters for scitypes
+
+Exports
+-------
+convert : dictionary of functions, with signature as below
+        indexed by triples of types
+         1st element = convert from - type
+         2nd element = convert to - type
+         3rd element = considered as this scitype - string
+        elements are conversion functions of machine type (1st) -> 2nd
+
+convert_to : function/dispatch version of convert, with 1st elt inferred
+
+
+-----------------------------------------
+
+Function signature of
+convert[(from_type, to_type, as_scitype)]
+
+Parameters
+----------
+what : from_type - object to convert
```

Comment:
strongly dislike name `what`.

Some alternatives: `object` (used in python docs for generic inputs, e.g. in `isinstance`), `data` (because I think we will always be using this function to convert data, as opposed to other kinds of things)

Is there a standard 'formal' name for the kind of thing that `y`, `X`, `Z` etc. are?

Reply:
changed to `obj` consistently, docstrings read much better now